--- GSimulation.cpp	2017-03-31 12:45:43.189485000 +0200
+++ GSimulation.cpp	2017-03-31 12:47:28.485826000 +0200
@@ -147,7 +147,9 @@
 //---------------------------------
 // TODO: Parallelize this region
 //---------------------------------
+  #pragma omp parallel
   {  // start of parallel region
+   #pragma omp parallel for schedule (dynamic) 
    for (i = 0; i < n; i++)// update acceleration
    {
      __assume_aligned(particles->pos_x, alignment);
@@ -188,7 +190,7 @@
      particles->acc_z[i] = az_i;
    }
    energy = 0;
-
+   #pragma omp for reduction(+:energy)
    for (i = 0; i < n; ++i)// update position
    {
      particles->vel_x[i] += particles->acc_x[i] * dt; //2flops
@@ -245,6 +247,8 @@
   dev=sqrt(dev/(double)(nf-2)-av*av);
   
   int nthreads=1;
+  #pragma omp parallel
+  nthreads=omp_get_num_threads(); 
 
   std::cout << std::endl;
   std::cout << "# Number Threads     : " << nthreads << std::endl;	   
