--- GSimulation.cpp	2017-03-28 10:06:25.798562000 +0200
+++ GSimulation.cpp	2017-03-28 10:24:11.988725000 +0200
@@ -147,7 +147,9 @@
 //---------------------------------
 // start of parallel region
 //---------------------------------
+  #pragma omp parallel
   {
+   #pragma omp for   
    for (i = 0; i < n; i++)// update acceleration
    {
     __assume_aligned(particles->pos_x, alignment);
@@ -189,7 +191,7 @@
 	particles->acc_z[i] = az_i;
    }
       energy = 0;
-
+      #pragma omp for reduction(+:energy)
       for (i = 0; i < n; ++i)// update position
       {
 		particles->vel_x[i] += particles->acc_x[i] * dt;	//2flops
@@ -246,6 +248,8 @@
   dev=sqrt(dev/(double)(nf-2)-av*av);
   
   int nthreads=1;
+  #pragma omp parallel
+  nthreads=omp_get_num_threads(); 
 
   std::cout << std::endl;
   std::cout << "# Number Threads     : " << nthreads << std::endl;	   
